module-03/assignments/assignment_GAURAVCHOUDHARY.md

```diff
@@ -0,0 +1,81 @@
+<!ATTLIST color_swatch image CDATA #REQUIRED>
+]>
+![image info](../assests/DTDValidate.png)
+
+
+
+3. Create XSD for this file and validate it using any of the tools we used
+Ans. 
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified"
+xmlns:xs="http://www.w3.org/2001/XMLSchema">
+<xs:element name="catalog">
+    <xs:complexType>
+        <xs:sequence>
+            <xs:element name="product">
+                <xs:complexType>
+                    <xs:sequence>
+                            <xs:element name="catalog_item" maxOccurs="unbounded" minOccurs="0">
+                                <xs:complexType>
+                                    <xs:sequence>
+                                        <xs:element type="xs:string" name="item_number"/>
+                                        <xs:element type="xs:float" name="price"/>
+                                        <xs:element name="size" maxOccurs="unbounded" minOccurs="0">
+                                            <xs:complexType>
+                                                <xs:sequence>
+                                                    <xs:element name="color_swatch" maxOccurs="unbounded" minOccurs="0">
+                                                        <xs:complexType>
+                                                            <xs:simpleContent>
+                                                                <xs:extension base="xs:string">
+                                                                    <xs:attribute type="xs:string" name="image" use="optional"/>
+                                                                </xs:extension>
+                                                            </xs:simpleContent>
+                                                        </xs:complexType>
+                                                    </xs:element>
+                                                </xs:sequence>
+                                                <xs:attribute type="xs:string" name="description" use="optional"/>
+                                            </xs:complexType>
+                                        </xs:element>
+                                    </xs:sequence>
+                                    <xs:attribute type="xs:string" name="gender" use="optional"/>
+                                </xs:complexType>
+                            </xs:element>
+                        </xs:sequence>
+                        <xs:attribute type="xs:string" name="product_id"/>
+                        <xs:attribute type="xs:string" name="description"/>
+                        <xs:attribute type="xs:string" name="product_image"/>
+                    </xs:complexType>
+                </xs:element>
+            </xs:sequence>
+        </xs:complexType>
+    </xs:element>
+</xs:schema>
+
+![image info](../assests/XSDValidate.png)
+
+
+
+
+4. Explain your thought process for these 2 declarations.
```

Comment:
#For DTD
While working with DTD I started with catalog then with has prodcut+. The + sign indicates that there are other elements inside that particular element. Now, the product has different attributes which are product description product_image, and product_id which are under CDATA which means that it is just a text and will not be parsed by the parser. Tags inside the text will not be expanded. After that we have catalog_item+ inside the product tag which has further attributes of Gender and is REQUIRED field. Furthermore, it has other tags which are item_number, price and size. And later everything else goes into succession just like the the above tags.

#For XSD
I followed the same strategy as DTD by going in chronological order. It has complex types which contains other elements. The simple content doesn't contain any other elements. It basically lets you define the structure and data  type of the XML document.